### PR TITLE
Different Techniques of Encoding Data for Transmission

### LINKED ISSUE #3323: Data encoding techniques
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of the article.
Data encoding techniques

### Introduction paragraph (2-3 paragraphs):
In a communication network, data is transmitted from the sender to the receiver. For this data to be transmitted properly it is encoded into binary digits 0s and 1s. 

This article intends to cover more on how data to be transmitted through a network is encoded. This therefore will emphasize more on different data encoding techniques. It also touches on Digital and analog data. 

The different data encoding techniques that shall be focused on include; Non-Return to Zero Level, Non-Return to Zero, Non-Return to Zero Inverted, Bi-phase encoding, and Block encoding. Data encoding types shall also be discussed.

### Key takeaways: 
A better understanding of data encoding.
A better understanding of different data encoding techniques e.g Non-Return to Zero Level(NRZ-L)
Be able to encode a data signal using any technique.
A better understanding of digital and analog signals.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by GeoffreyKangu**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3323

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Qodestackr** [comment]
Hi I'll be your peer reviewer

**Comment by Qodestackr** [comment]
@254-geo I will be taking you through a preliminary review.

File: content/articles/different-techniques-of-encoding-data-for-transmission/index.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+### Non-Return to Zero
+In this encoding technique;
+High voltages are represented by 1s.
+Low voltages are represented by 0s.
+During bit interval, voltages remain the main.
+
+This is the main behavior of this encoding technique.
+There is no indication for the beginning and the ending of a bit. Due to no voltage change during bit interval, there is no difference in the value of the previous and present bit.
+![](/engineering-education/different-techniques-of-encoding-data-for-transmission/nrz.png)
+
```
Comment (Qodestackr) [inline content/articles/different-techniques-of-encoding-data-for-transmission/index.md:35]: Make sure to add alt text to all your image paths.

**Comment by Qodestackr** [comment]
Going through your article, I have noticed that this article has an abrupt ending. The length of the article does not seem to involve the reader through the concepts. Make sure the ending isn't abrupt. Could you consider adding the pros and cons?

File: content/articles/different-techniques-of-encoding-data-for-transmission/index.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+####	Bi-phase Manchester
+Here, transition is done in the middle of the bit interval.
+Resultant pulse of the transition is from high to low when input is 1 in the middle and vice versa for input 0.
+
+The figure below shows the waveforms of NRZ-L, NRZ-I, Bi-phase Manchester and Differential Manchester coding for different digital inputs.
+
+![](/engineering-education/different-techniques-of-encoding-data-for-transmission/waveforms.png)
+### Block encoding.
+Includes 4B/5B and 8B/6T encoding techniques.
```
Comment (Qodestackr) [inline content/articles/different-techniques-of-encoding-data-for-transmission/index.md:70]: This is too short.

File: content/articles/different-techniques-of-encoding-data-for-transmission/index.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+### Introduction
+In a communication network, data is transmitted from the sender to the receiver. For this data to be transmitted properly it is encoded into binary digits 0s and 1s.
+
+This article intends to cover more on how data to be transmitted through a network is encoded. This therefore will emphasize more on different data encoding techniques. It will also talk about on digital and analog data.
+
+The different data encoding techniques that shall be focused on include; Non-Return to Zero Level, Non-Return to Zero, Non-Return to Zero Inverted, Bi-phase encoding, and Block encoding.
+
+Data encoding types shall also be discussed.
+For this data to be encoded it has to be converted to signal variations. This signal variations include;
```
Comment (Qodestackr) [inline content/articles/different-techniques-of-encoding-data-for-transmission/index.md:9]: Make sure to format your intro section to improve readability. It seems to be technical for beginners.
Ping me when done :+1:

**Review by Qodestackr — CHANGES_REQUESTED** [review]

**Comment by GeoffreyKangu** [comment]
> Going through your article, I have noticed that this article has an abrupt ending. The length of the article does not seem to involve the reader through the concepts. Make sure the ending isn't abrupt. Could you consider adding the pros and cons?

Changes made @Qodestackr

**Comment by Qodestackr** [comment]
Glad you made changes. Kindly run the article through Grammarly once more so that I can head over to the next review

**Comment by hectorkambow** [comment]
@GeoffreyKangu 

[Different Techniques of Encoding Data for Transmission #3411.pdf](https://github.com/section-engineering-education/engineering-education/files/7027870/Different.Techniques.of.Encoding.Data.for.Transmission.3411.pdf)


👋 Good afternoon - and welcome to the EngEd program.
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by Qodestackr** [comment]
@GeoffreyKangu kindly fix pliagiarism as well as I proceed with further reviews. 👌

**Comment by GeoffreyKangu** [comment]
@hectorkambow  Hello I seen the plagiarism and am working on it . Thank you.

**Comment by GeoffreyKangu** [comment]
@Qodestackr  Hello am working on it I will let you when done

**Comment by GeoffreyKangu** [comment]
@Qodestackr   Hello, I have made changes.

**Comment by hectorkambow** [comment]
[Different Techniques of Encoding Data for Transmission #3411 2nd.pdf](https://github.com/section-engineering-education/engineering-education/files/7080154/Different.Techniques.of.Encoding.Data.for.Transmission.3411.2nd.pdf)


% is around the same.

**Comment by hectorkambow** [comment]
Plagiarism check done

**Review by Kayere — CHANGES_REQUESTED** [review]
Hello @GeoffreyKangu, thank you for submitting this PR. I have done a review on  your article and I feel like there is more to cover in this topic. Could you please add more content on your article so as to add value. You could add some real examples of how data can be encoded using the techniques you have mentioned.

Also, make sure your article is unique. Please make the changes and ping me when you are done. Thanks.

**Review by Kayere — APPROVED** [review]
Congratulations for publishing your first article @GeoffreyKangu. Keep up the good work.
Thank you for the review @Qodestackr.

**Pull request merged**